src/chains/algorand_1/algoTransaction.ts

```diff
@@ -267,25 +299,51 @@ export class AlgorandTransaction implements Transaction {
     }
   }
 
-  /** Returns address, for which, a matching signature must be attached to transaction
-   *  Throws if actions[].from is not a valid address - needed to determine the required signature */
+  /** Returns address, for which, a matching signature must be attached to transaction */
   public get missingSignatures(): AlgorandAddress[] {
     this.assertIsValidated()
-    const missingSignature = this.hasAllRequiredSignatures ? null : this.requiredAuthorization
-    return isNullOrEmpty(missingSignature) ? null : [missingSignature] // if no values, return null instead of empty array
+    const missingSignatures = this.requiredAuthorizations?.filter(
+      auth => !this.hasSignatureForPublicKey(getAlgorandPublicKeyFromAddress(auth)),
+    )
+
+    /** check if number of signatures present are greater then or equal to multisig threshold.
+     * If so, set missing signatures to null */
+    if (this.multiSigOptions) {
+      return this.multiSigOptions.addrs.length - (this.multiSigOptions.threshold as number) <= missingSignatures.length
+        ? null
+        : missingSignatures
+    }
+
+    return isNullOrEmpty(missingSignatures) ? null : missingSignatures // if no values, return null instead of empty array
   }
 
-  /** Returns address specified by actions[].from property
-   * throws if actions[].from is not a valid address - needed to determine the required signature */
+  /** Returns the required addresses for a transaction/multisig transaction */
   public get requiredAuthorizations(): AlgorandAddress[] {
-    return [this.requiredAuthorization]
+    return this.requiredAuthorization
   }
 
   /** private property for the one signature address required (by action.from) */
-  private get requiredAuthorization(): AlgorandAddress {
+  private get requiredAuthorization(): AlgorandAddress[] {
     this.assertIsValidated()
     this.assertFromIsValidAddress()
-    return this.action.from
+    if (this.multiSigOptions) {
+      return this.fetchRequiredAuthsForMultiSigTransaction()
+    }
+    return [this.action.from]
```

Comment:
if this.action.from is null or undefined this will return an array with a undefined member. Instead you want an empty array to be returned if this.action.from is null... i.e.
return (this.action.from) ? [this.action.from] : []

Reply:
done

Reply:
done